nrf52-hal-common/src/spim.rs

```diff
@@ -270,105 +251,153 @@ impl<T> Spim<T> where T: SpimExt {
     /// Read from an SPI slave
     ///
+    /// This method is deprecated. Consider using `transfer` or `transfer_split`
+    #[inline(always)]
+    pub fn read(
+        &mut self,
+        chip_select: &mut Pin<Output<PushPull>>,
+        tx_buffer: &[u8],
+        rx_buffer: &mut [u8],
+    ) -> Result<(), Error> {
+        self.transfer_split_uneven(chip_select, tx_buffer, rx_buffer)
+    }
+
+    /// Read and write from a SPI slave, using a single buffer
+    ///
     /// This method implements a complete read transaction, which consists of
     /// the master transmitting what it wishes to read, and the slave responding
     /// with the requested data.
     ///
     /// Uses the provided chip select pin to initiate the transaction. Transmits
-    /// all bytes in `tx_buffer`, then receives bytes until `rx_buffer` is full.
-    /// Both buffer must have a length of at most 255 bytes.
-    pub fn read(&mut self,
+    /// all bytes in `buffer`, then receives an equal number of bytes.
+    pub fn transfer(
+        &mut self,
         chip_select: &mut Pin<Output<PushPull>>,
-        tx_buffer  : &[u8],
-        rx_buffer  : &mut [u8],
-    )
-        -> Result<(), Error>
-    {
-        if tx_buffer.len() > EASY_DMA_SIZE {
-            return Err(Error::TxBufferTooLong);
-        }
-        if rx_buffer.len() > EASY_DMA_SIZE {
-            return Err(Error::RxBufferTooLong);
-        }
-
-        self.do_spi_dma_transfer(
-            // Set up the DMA write
-            // We're giving the register a pointer to the stack. Since we're
-            // waiting for the SPI transaction to end before this stack pointer
-            // becomes invalid, there's nothing wrong here.
-            //
-            // The PTR field is a full 32 bits wide and accepts the full range
-            // of values.
-
-            // tx_data_ptr:
-            tx_buffer.as_ptr() as u32,
-
-            // We're giving it the length of the buffer, so no danger of
-            // accessing invalid memory. We have verified that the length of the
-            // buffer fits in an `u8`, so the cast to the type of maxcnt
-            // is also fine.
-            //
-            // Note that that nrf52840 maxcnt is a wider
-            // type than a u8, so we use a `_` cast rather than a `u8` cast.
-            // The MAXCNT field is thus at least 8 bits wide and accepts the full
-            // range of values that fit in a `u8`.
-
-            // tx_len:
-            tx_buffer.len() as _,
-
-            // Set up the DMA read
-            // This is safe for the same reasons that writing to TXD.PTR is
-            // safe. Please refer to the explanation there.
+        buffer: &mut [u8],
+    ) -> Result<(), Error> {
+        ram_slice_check(buffer)?;
+
+        buffer.chunks(EASY_DMA_SIZE).try_for_each(|chunk| {
+            self.do_spi_dma_transfer(
+                DmaSlice::from_slice(chunk),
+                DmaSlice::from_slice(chunk),
+                |cs| {
```

Comment:
Also not here. Do we want to do this ever?

Reply:
Both to toggle CS and leave it low while waiting to restart should be ok.